File: okj/szoftverfejleszto/kutyak/c#/Kutya.cs
using System;

public class Kutya {
    public readonly int id;
    public readonly int fajtaId;
    public readonly int nevId;
    public readonly int eletkor;
    public readonly DateTime ellenorzes;

    public Kutya(string[] data) {
        id = int.Parse(data[0]);
        fajtaId = int.Parse(data[1]);
        nevId = int.Parse(data[2]);
        eletkor = int.Parse(data[3]);
        ellenorzes = DateTime.ParseExact(data[4], "yyyy.MM.dd", System.Globalization.CultureInfo.InvariantCulture);
    }
}

File: okj/szoftverfejleszto/kutyak/java/Kutya.java
import java.time.*;
import java.time.format.*;

public class Kutya {
    private static final DateTimeFormatter ellenorzesDateFormat = DateTimeFormatter.ofPattern("yyyy.MM.dd");

    public final int id;
    public final int fajtaId;
    public final int nevId;
    public final int eletkor;
    public final LocalDate ellenorzes;

    public Kutya(String[] data) {
        id = Integer.parseInt(data[0]);
        fajtaId = Integer.parseInt(data[1]);
        nevId = Integer.parseInt(data[2]);
        eletkor = Integer.parseInt(data[3]);
        ellenorzes = LocalDate.parse(data[4], ellenorzesDateFormat);
    }
}